### PR TITLE
[WIP] Allow Proxmox Snapshot Restoring

### PR DESCRIPTION
**Pull request by EinDev**
##### SUMMARY
Allows the restoration of a proxmox snapshot

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
This Change allows restoring a snapshot in Proxmox

##### ADDITIONAL INFORMATION
I hope this is the correct branch...

### CODE REVIEW

File: changelogs/fragments/4375-allow-proxmox-snapshot-restoring.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox - add restore snapshot option
```
Comment (felixfontein) [inline changelogs/fragments/4375-allow-proxmox-snapshot-restoring.yml:2]:
```suggestion
  - proxmox - add restore snapshot option (https://github.com/ansible-collections/community.general/pull/4375).
```

File: plugins/modules/cloud/misc/proxmox_snap.py [hunk 2]
```diff
@@ -28,7 +29,7 @@
   state:
     description:
      - Indicate desired state of the instance snapshot.
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_snap.py:31]: You also want to add some documentation on how that value works.
```suggestion
     - Indicate desired state of the instance snapshot.
     - The C(rollback) value was added in community.general 4.7.0.
```
You should also mention in the documentation of `snapname` that it is required if `I(state=rollback)`.

File: plugins/modules/cloud/misc/proxmox_snap.py [hunk 3]
```diff
@@ -144,7 +176,7 @@ def main():
         vmid=dict(required=False),
         hostname=dict(),
         timeout=dict(type='int', default=30),
-        state=dict(default='present', choices=['present', 'absent']),
+        state=dict(default='present', choices=['present', 'absent', 'rollback']),
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_snap.py:179]: You probably want to add `require_if=[('state', 'rollback', ('snapname', ))],` to the `AnsibleModule()` call below.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I added some first comments.

Please rebase and retarget your PR so that it is against the `main` branch. All PRs must land in `main` (except very few exceptions).

**Comment by EinDev** [comment]
Thank you @felixfontein for your helpful comments. As i don't know how to rebase and retarget the PR (and i have actually "accidentally" already done it on `main` - don't ask) i will close this PR and create a new one.

**Pull request closed without merging**